core-runner/src/runner.rs

```diff
@@ -335,6 +335,7 @@ impl<MQ: MessageQueue, PS: ProgramStorage, WL: WaitList, E: Environment<Ext>>
 
         let mut context = self.create_context(allocations);
         let next_message_id = next_message.id();
+        let next_message_exit_code = next_message.reply().map(|v| v.1).unwrap_or(0);
```

Comment:
Should add a comment here to explain that we don't generate error-message while handling replies